File: Tests/python/nine.py
class InventoryManager:

	def __init__(self):
		self.items = [
			{"id": 1, "name": "Apple", "spoiled": True},
			{"id": 2, "name": "Bread", "spoiled": True},
			{"id": 3, "name": "Sword", "spoiled": False},
			{"id": 4, "name": "Meat", "spoiled": True}
		]

	def remove_spoiled_items(self):
		for item in self.items:
			if item.get("spoiled") is True:
				print(f"Metam ārā: {item['name']}")
				self.items.remove(item)


if __name__ == "__main__":
	inventory = InventoryManager()
	inventory.remove_spoiled_items()

	spoiled_remaining = [i for i in inventory.items if i["spoiled"]]
	assert len(spoiled_remaining) == 0, f"Kļūda: Ne visi bojātie priekšmeti izdzēsti! Palika: {spoiled_remaining}"
	print("Scenārijs 9 darbojas veiksmīgi.")

File: Tests/c/nine.c
#include <stdio.h>
#include <assert.h>

typedef struct {
    int id;
    int spoiled;
} Item;

typedef struct {
    Item items[10];
    int count;
} Inventory;

void remove_spoiled(Inventory *inv) {
    for (int i = 0; i < inv->count; i++) {
        if (inv->items[i].spoiled) {
            for (int j = i; j < inv->count - 1; j++) {
                inv->items[j] = inv->items[j + 1];
            }
            inv->count--;
        }
    }
}

int main() {
    Inventory inv;
    inv.count = 4;
    inv.items[0] = (Item){1, 1};
    inv.items[1] = (Item){2, 1};
    inv.items[2] = (Item){3, 0};
    inv.items[3] = (Item){4, 1};

    remove_spoiled(&inv);

    // Pārbaudām, vai palika kāds bojāts priekšmets
    int has_spoiled = 0;
    for (int i = 0; i < inv.count; i++) {
        if (inv.items[i].spoiled) has_spoiled = 1;
    }

    assert(has_spoiled == 0 && "Kļūda: Ne visi bojātie priekšmeti tika izdzēsti!");
    printf("Scenārijs 9 darbojas veiksmīgi.\n");
    return 0;
}